File: advent-of-code/2015/03/01/solution.cs
using System.Diagnostics;

static int CountHouses(string directions)
{
    var x = 0;
    var y = 0;
    var visited = new HashSet<string> { "0,0" };
    
    foreach (var c in directions)
    {
        switch (c)
        {
            case '^': y++; break;
            case 'v': y--; break;
            case '>': x++; break;
            case '<': x--; break;
        }
        visited.Add($"{x},{y}");
    }
    
    return visited.Count;
}

// Test cases with assertions
Debug.Assert(CountHouses(">") == 2);
Debug.Assert(CountHouses("^>v<") == 4);
Debug.Assert(CountHouses("^v^v^v^v^v") == 2);

// Read input and calculate result
var inputPath = Path.Combine("..", "input.txt");
var directions = File.ReadAllText(inputPath).Trim();

var result = CountHouses(directions);
Console.WriteLine($"C#: {result}");

File: advent-of-code/2015/03/01/solution.py
import os

def count_houses(directions):
    x, y = 0, 0
    visited = {(0, 0)}
    
    for char in directions:
        if char == '^':
            y += 1
        elif char == 'v':
            y -= 1
        elif char == '>':
            x += 1
        elif char == '<':
            x -= 1
        visited.add((x, y))
    
    return len(visited)

# Test cases with assertions
assert count_houses(">") == 2
assert count_houses("^>v<") == 4
assert count_houses("^v^v^v^v^v") == 2

# Read input and calculate result
script_dir = os.path.dirname(os.path.abspath(__file__))
input_path = os.path.join(script_dir, "..", "input.txt")
with open(input_path) as f:
    directions = f.read().strip()

result = count_houses(directions)
print(f"Python: {result}")
